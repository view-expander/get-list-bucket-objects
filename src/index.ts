import * as AWS from 'aws-sdk'
import { APIGatewayProxyResult } from 'aws-lambda'
const s3 = new AWS.S3()

export async function handler(): Promise<APIGatewayProxyResult> {
  try {
    const Bucket = process.env.BUCKET

    if (Bucket === undefined) {
      throw new Error('Bucket name is required')
    }

    const res = await s3.listObjectsV2({ Bucket, Prefix: 'source/' }).promise()

    return {
      statusCode: res.$response.httpResponse.statusCode,
      body: JSON.stringify(
        (res.Contents || []).filter(
          (item) => item.Key && /^.+\.jpe?g$/i.test(item.Key)
        )
      ),
    }
  } catch (err) {
    return {
      statusCode: err.statusCode || 400,
      body: err.message,
    }
  }
}
