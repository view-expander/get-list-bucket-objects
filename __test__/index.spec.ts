import * as AWS from 'aws-sdk'
import { handler } from '../src'

describe('handler()', () => {
  beforeEach(() => {
    // set environments
    process.env.BUCKET = undefined

    // mock aws-sdk
    jest
      .spyOn((AWS.S3 as any).services['2006-03-01'].prototype, 'listObjectsV2')
      .mockImplementation(() => ({
        promise: (): any => ({
          $response: {
            httpResponse: {
              statusCode: 200,
            },
          },
          Contents: [
            {
              Key: 'hoge.jpg',
              LastModified: '2020-05-09T06:33:51.000Z',
              ETag: '44c452ca6297705b5c7bea4246ecaf09',
              Size: 2636121,
              StorageClass: 'StorageClass',
            },
            {
              Key: 'piyo.svg',
              LastModified: '2020-05-09T06:33:51.000Z',
              ETag: '37305f4d32679861ad71971c843ddbeb',
              Size: 1927537,
              StorageClass: 'StorageClass',
            },
            {
              Key: 'fuga.jpg',
              LastModified: '2020-05-09T06:33:50.000Z',
              ETag: '2e7d2444cbb4e79680a64288eeaa7d5e',
              Size: 2074755,
              StorageClass: 'StorageClass',
            },
          ],
          Name: 'TARGET_BUCKET_NAME',
          Prefix: 'source/',
          MaxKeys: 100,
          CommonPrefixes: [],
          KeyCount: 100,
          NextContinuationToken: 'DUMMY_TOKEN',
        }),
      }))
  })

  afterEach(() => {
    jest.restoreAllMocks()
  })

  it('be succeed', async () => {
    expect.assertions(1)

    // set environments
    process.env.BUCKET = 'TARGET_BUCKET_NAME'

    return expect(
      handler({
        queryStringParameters: null,
      } as any)
    ).resolves.toMatchSnapshot()
  })
})
