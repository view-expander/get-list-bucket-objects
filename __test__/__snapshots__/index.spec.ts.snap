// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handler() be succeed 1`] = `
Object {
  "body": "[{\\"Key\\":\\"hoge.jpg\\",\\"LastModified\\":\\"2020-05-09T06:33:51.000Z\\",\\"ETag\\":\\"44c452ca6297705b5c7bea4246ecaf09\\",\\"Size\\":2636121,\\"StorageClass\\":\\"StorageClass\\"},{\\"Key\\":\\"piyo.jpg\\",\\"LastModified\\":\\"2020-05-09T06:33:51.000Z\\",\\"ETag\\":\\"37305f4d32679861ad71971c843ddbeb\\",\\"Size\\":1927537,\\"StorageClass\\":\\"StorageClass\\"},{\\"Key\\":\\"fuga.jpg\\",\\"LastModified\\":\\"2020-05-09T06:33:50.000Z\\",\\"ETag\\":\\"2e7d2444cbb4e79680a64288eeaa7d5e\\",\\"Size\\":2074755,\\"StorageClass\\":\\"StorageClass\\"}]",
  "statusCode": 200,
}
`;
